core/src/main/java/apoc/export/cypher/MultiStatementCypherSubGraphExporter.java

```diff
@@ -102,7 +104,14 @@ public void export(ExportConfig config, Reporter reporter, ExportFileManager cyp
                 exportRelationships(relationshipsWriter, reporter, batchSize);
                 break;
             default:
-                artificialUniques += countArtificialUniques(graph.getNodes());
+                artificialUniqueNodes += countArtificialUniqueNodes(graph.getNodes());
+                // In this code path an artificial relationship property is added for each relationship
+                if (exportConfig.isMultipleRelationshipsWithType()) {
+                    artificialUniqueRels += StreamSupport.stream(
+                                    graph.getRelationships().spliterator(), false)
+                            .toList()
+                            .size();
```

Comment:
No need to materialize the relationships if we only want to count them. 

```suggestion
                    artificialUniqueRels += StreamSupport.stream(
                                    graph.getRelationships().spliterator(), false)
                            .count();
```